scripts/enforce-call-super-for-overrided-onupdate

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+function is_call_super_found {
+  # If class directly extends BaseProjectKeyAvailabilityAssuranceAction then call super.onUpdate is no needed
+  grep "extends BaseProjectKeyAvailabilityAssuranceAction" "$0" > /dev/null
+  if [[ $? -eq 0 ]]
+  then
+    return 0
+  fi
+
+  # If no call super.onUpdate found then return 255 that cause xargs failure
+  grep "super.onUpdate" "$0" > /dev/null
+  if [[ $? -ne 0 ]]
+  then
+    echo Call 'super.onUpdate' not found in "$0"
+    return 255
+  fi
+}
+
+export -f is_call_super_found
+
+git grep --files-with-matches "void onUpdate" | grep --invert-match "BaseProjectKeyAvailabilityAssuranceAction.java" | xargs -n 1 bash -c "is_call_super_found $@" 2> /dev/null
```

Comment:
Note that `git grep --files-with-matches "void onUpdate"` includes this very script as well, possibly an `:!scripts/` would be useful

Reply:
Same for `grep --invert-match "BaseProjectKeyAvailabilityAssuranceAction.java"`, can be just replaced with `:!*/BaseProjectKeyAvailabilityAssuranceAction.java`

Reply:
`xargs -n 1 bash` results in ~10 bash processes getting spawned (one for each file)... possibly this can be simplified to just `xargs git grep --files-without-match 'extends BaseProjectKeyAvailabilityAssuranceAction\|super.onUpdate'` (and then the auxiliary function isn't necessary)

Reply:
Ok, bo how to join `':!*/BaseProjectKeyAvailabilityAssuranceAction.java'` and `':!scripts/'`?
They can't be regex IIRC

Reply:
Hmm doesn't
```
git grep --files-with-matches 'void onUpdate' -- ':!scripts/' ':!*/BaseProjectKeyAvailabilityAssuranceAction.java'```
work?

Reply:
Hmm, strange: I was sure that I tested this and it didn't work! But yes, it works, thanks!